src/Microsoft.DotNet.Build.Tasks/PrereleaseResolveNuGetPackageAssets.cs

```diff
@@ -669,22 +670,239 @@ private static string TryGetRuntimeVersion(string path)
             uint actualBufferSize = 0;
             int bufferLength = 11; // 11 is the length of a runtime version and null terminator v2.0.50727/0
 
-            do
+            try
             {
-                runtimeVersion = new StringBuilder(bufferLength);
-                hresult = NativeMethods.GetFileVersion(path, runtimeVersion, bufferLength, out actualBufferSize);
-                bufferLength = bufferLength * 2;
+                do
+                {
+                    runtimeVersion = new StringBuilder(bufferLength);
+                    hresult = NativeMethods.GetFileVersion(path, runtimeVersion, bufferLength, out actualBufferSize);
+                    bufferLength = bufferLength * 2;
 
-            } while (hresult == NativeMethods.ERROR_INSUFFICIENT_BUFFER);
+                } while (hresult == NativeMethods.ERROR_INSUFFICIENT_BUFFER);
 
-            if (hresult == NativeMethods.S_OK && runtimeVersion != null)
+                if (hresult == NativeMethods.S_OK && runtimeVersion != null)
+                {
+                    return runtimeVersion.ToString();
+                }
+                else
+                {
+                    return String.Empty;
+                }
+            }
+            catch (EntryPointNotFoundException)
             {
-                return runtimeVersion.ToString();
+                return ManagedRuntimeVersionReader.GetRuntimeVersion(path);
             }
-            else
+        }
+    }
+
+    /// <summary>
+    /// Managed implementation of a reader for getting the runtime version of an assembly
+    /// </summary>
+    static class ManagedRuntimeVersionReader
```

Comment:
@ellismg back when this code was added to MSBuild you said there's some stuff missing from PE Reader that needs to be exposed: https://github.com/Microsoft/msbuild/pull/43#issuecomment-88251329